OTAnalytics/application/plotting.py

```diff
@@ -1,12 +1,78 @@
-from typing import Optional
+from abc import abstractmethod
+from typing import Callable, Optional
 
 from OTAnalytics.application.datastore import Datastore
-from OTAnalytics.application.state import Plotter, TrackViewState
+from OTAnalytics.application.state import ObservableProperty, Plotter, TrackViewState
 from OTAnalytics.domain.track import TrackImage
 
 
+class Layer:
+    @abstractmethod
+    def get_name(self) -> str:
+        """Get the name of the layer.
+
+        Returns:
+            str: the name.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def enable(self) -> None:
+        """Enables the layer."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def disable(self) -> None:
+        """Disables the layer."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_enabled(self) -> bool:
+        """Returns wether layer is enabled."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def register(self, observer: Callable[[bool], None]) -> None:
+        """Register observer to layer to get notifications on layer enabled state
+        changes.
+        """
+        raise NotImplementedError
+
+
+class PlottingLayer(Plotter, Layer):
+    def __init__(self, name: str, other: Plotter, enabled: bool) -> None:
+        self._name = name
+        self._other = other
+        self._enabled: ObservableProperty[bool] = ObservableProperty[bool](enabled)
+
+    def plot(self) -> Optional[TrackImage]:
+        """Plots layer if enabled.
+
+        Returns:
+            Optional[TrackImage]: the image that the layer has been plotted onto.
+        """
+        return self._other.plot() if self._enabled.get() else None
+
+    def get_name(self) -> str:
+        return self._name
+
+    def enable(self) -> None:
+        if not self._enabled.get():
+            self._enabled.set(True)
+
+    def disable(self) -> None:
+        if self._enabled.get():
+            self._enabled.set(False)
```

Comment:
We can merge this into a `set_enabled` method. The observable property will already check if the value changes.